rest_framework/authentication.py

```diff
@@ -167,31 +167,19 @@ def get_model(self):
         from rest_framework.authtoken.models import Token
         return Token
 
-    """
-    A custom token model may be used, but must have the following properties.
-
-    * key -- The string identifying the token
-    * user -- The user to which the token belongs
-    """
-
     def authenticate(self, request):
         auth = get_authorization_header(request).split()
 
-        if not auth or auth[0].lower() != self.keyword.lower().encode():
+        if len(auth) != 2 or auth[0].lower() != self.keyword.lower().encode():
             return None
-
-        if len(auth) == 1:
-            msg = _('Invalid token header. No credentials provided.')
-            raise exceptions.AuthenticationFailed(msg)
-        elif len(auth) > 2:
-            msg = _('Invalid token header. Token string should not contain spaces.')
-            raise exceptions.AuthenticationFailed(msg)
```

Comment:
What happened to these error messages? The idea is to validate the shape of the token header; with your proposed code, an invalid token will go unnoticed.